src/services/PostService.ts

```diff
@@ -131,6 +131,15 @@ export class PostService {
     });
   }
 
+  public async getSavedPostsByUserId(user: UserModel): Promise<PostModel[]> {
+    return this.transactions.readOnly(async (transactionalEntityManager) => {
+      const userRepository = Repositories.user(transactionalEntityManager);
+      const userWithSaved = await userRepository.getSavedPostsByUserId(user.id);
+      if (!userWithSaved) throw new NotFoundError('Easter egg')
```

Comment:
easter egg? lmao, more like you didn't properly set this field when a user was created or you have old data that needs to have something set for this field; is this a meme or is this actually handling anything?

Reply:
What is happening is that we get the user from the session token and pass that user through as a parameter, but that user does not include its saved posts. We then need to query the user by its id and leftjoin its saved posts but we already would have thrown an error if the user was not properly logged in with a valid id so it should never be possible for that user to not be found. But this is necessary because otherwise the type being passed through remains ( UserModel | undefined ) but we can rule out undefined by this if statement just that it should never actually evaluate to true.

Reply:
I see, this seems to be something throughout this codebase. Since you know that you should be expecting a user given they're logged in already due to your use of the CurrentUser decorator, you should force unwrap `userWithSaved`. Take a look at https://spin.atomicobject.com/2022/03/28/optional-undefined-typescript/ to see how optionals work and consider doing a refactor in the future converting all `SomeType | undefined` instances to `SomeType?`